### PR TITLE
first swing at ORM

### PR DESCRIPTION
**Pull request by karcass**
feedback appreciated

### CODE REVIEW

File: packages/mds-transactions-service/repository/entities/operation-entity.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+@Entity('operations')
+export class TransactionOperationEntity
+  extends IdentityColumn(RecordedColumn(class {}))
+  implements TransactionOperationDomainModel {
+  @Column('uuid', { primary: true })
+  operation_id: TransactionOperationDomainModel['operation_id']
+
+  @Column('uuid', { primary: true })
```
Comment (janedotx) [inline packages/mds-transactions-service/repository/entities/operation-entity.ts:20]: I notice `operation_id` and `transaction_id` are both primary columns. That's probably not right, right?

File: packages/mds-transactions-service/repository/entities/operation-entity.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  transaction_id: TransactionOperationDomainModel['transaction_id']
+
+  @Column('timestamp')
+  timestamp: TransactionOperationDomainModel['timestamp']
+
+  @Column('varchar')
+  operation_type: TransactionOperationDomainModel['operation_type']
+
+  @Column('varchar')
```
Comment (janedotx) [inline packages/mds-transactions-service/repository/entities/operation-entity.ts:29]: Change the `varchar` on 29 to `jsonb`.

File: packages/mds-transactions-service/repository/entities/status-entity.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  @Column('uuid', { primary: true })
+  transaction_id: TransactionStatusDomainModel['transaction_id']
+
+  @Column('timestamp')
+  timestamp: TransactionStatusDomainModel['timestamp']
+
+  @Column('varchar')
+  status_type: TransactionStatusDomainModel['status_type']
+
+  @Column('varchar')
```
Comment (janedotx) [inline packages/mds-transactions-service/repository/entities/status-entity.ts:29]: Change this to `jsonb`.

**Review by janedotx — CHANGES_REQUESTED** [review]

File: packages/mds-transactions-service/repository/entities/transaction-entity.ts [hunk 4]
```diff
@@ -3,15 +3,35 @@ import { IdentityColumn, RecordedColumn } from '@mds-core/mds-repository'
...
 
   @Column('varchar')
-  text: TransactionEntityModel['text']
+  fee_type: TransactionEntityModel['fee_type']
+
+  @Column('int')
+  amount: TransactionEntityModel['amount'] // pennies
+
+  @Column('json')
+  receipt: TransactionEntityModel['receipt'] // is this how you specify a JSON blob?
```
Comment (drtyh2o) [inline packages/mds-transactions-service/repository/entities/transaction-entity.ts:36]: Yes, it is. Comment not necessary.

**Pull request closed without merging**